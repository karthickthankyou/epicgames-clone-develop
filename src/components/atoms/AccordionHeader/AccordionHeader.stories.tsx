import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import AccordionHeader from './AccordionHeader'

export default {
  title: 'atoms/AccordionHeader',
  component: AccordionHeader,
} as ComponentMeta<typeof AccordionHeader>

const Template: ComponentStory<typeof AccordionHeader> = ({
  name,
  open,
  setOpen,
}) => <AccordionHeader name={name} open={open} setOpen={setOpen} />

export const Primary = Template.bind({})
Primary.args = {
  name: 'Hello',
  open: false,
  setOpen: () => {},
}
Primary.parameters = {}
