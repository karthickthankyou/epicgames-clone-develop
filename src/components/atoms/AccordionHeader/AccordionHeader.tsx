import { SetStateAction, Dispatch } from 'react'
import { ChevronUpIcon, ChevronDownIcon } from 'src/assets'

export interface IAccordionHeaderProps {}

const AccordionHeader = ({
  name,
  open,
  setOpen,
}: {
  name: string
  open: boolean
  setOpen: Dispatch<SetStateAction<boolean>>
}) => (
  <button
    type='button'
    className={` flex justify-between w-full p-4 tracking-widest text-left text-gray-300 uppercase ${
      open && 'bg-gray-800 '
    }`}
    onClick={() => setOpen((state) => !state)}
  >
    {name}
    {open ? (
      <ChevronUpIcon className='w-4 h-4 ml-auto text-gray-200' />
    ) : (
      <ChevronDownIcon className='w-4 h-4 ml-auto text-gray-500' />
    )}
  </button>
)

export default AccordionHeader
