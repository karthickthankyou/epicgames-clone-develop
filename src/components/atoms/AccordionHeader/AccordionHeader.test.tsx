import React from 'react'
import { render } from '@testing-library/react'
import AccordionHeader from './AccordionHeader'

describe('AccordionHeader Component', () => {
  const setSelectedProperty = jest.fn()
  test('it should match the snapshot', () => {
    const { asFragment } = render(
      <AccordionHeader
        name='Karthick'
        open={false}
        setOpen={setSelectedProperty}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
