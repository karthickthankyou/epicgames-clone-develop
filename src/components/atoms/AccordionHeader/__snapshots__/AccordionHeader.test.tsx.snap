// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionHeader Component it should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class=" flex justify-between w-full p-4 tracking-widest text-left text-gray-300 uppercase false"
    type="button"
  >
    Karthick
    <svg
      class="w-4 h-4 ml-auto text-gray-500"
    >
      chevron-down.svg
    </svg>
  </button>
</DocumentFragment>
`;
